tests/DriverManagerTest.php

```diff
@@ -149,6 +149,51 @@ public function testDatabaseUrlPrimaryReplica(): void
         );
     }
 
+    public function testDatabaseUrlPrimaryReplicaWithServerVersionInUrl(): void
+    {
+        $options = [
+            'driver' => 'pdo_mysql',
+            'primary' => ['url' => 'mysql://foo:bar@localhost:11211/baz?serverVersion=11.2'],
+            'replica' => [
+                'replica1' => ['url' => 'mysql://foo:bar@localhost:11211/baz_replica?serverVersion=11.2'],
+            ],
+            'wrapperClass' => PrimaryReadReplicaConnection::class,
+        ];
+
+        $conn = DriverManager::getConnection($options);
+
+        $params = $conn->getParams();
+        self::assertInstanceOf(PDO\MySQL\Driver::class, $conn->getDriver());
+
+        $expected = [
+            'user'          => 'foo',
+            'password'      => 'bar',
+            'host'          => 'localhost',
+            'port'          => 11211,
+            'dbname'        => 'baz',
+            'driver'        => 'pdo_mysql',
+            'url'           => 'mysql://foo:bar@localhost:11211/baz?serverVersion=11.2',
+            'serverVersion' => '11.2',
+        ];
+
+        self::assertEquals(
+            [
+                'primary' => $expected,
+                'replica' => [
+                    'replica1' => array_merge(
+                        $expected,
+                        [
+                            'dbname' => 'baz_replica',
+                            'url'    => 'mysql://foo:bar@localhost:11211/baz_replica?serverVersion=11.2',
+                        ]
+                    ),
+                ],
+                'serverVersion' => '11.2',
+            ],
+            array_intersect_key($params, ['primary' => null, 'replica' => null, 'serverVersion' => null])
```

Comment:
Please [refrain from putting logic into tests](https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html). Instead, provide the expected values statically.